File: code/jvm/src/main/kotlin/pt/isel/leic/ptgest/domain/common/model/BodyData.kt
package pt.isel.leic.ptgest.domain.common.model

import pt.isel.leic.ptgest.domain.common.SkinFold

data class BodyData(
    val weight: Double,
    val height: Double,
    val bodyCircumferences: BodyCircumferences,
    val bodyComposition: BodyComposition,
    val skinFolds: Map<SkinFold, Double>? = null
)

File: code/js/src/views/user/TrainerViews/models/traineeData/BodyData.ts
import BodyCircumferences from "./BodyCircumferences.ts"
import BodyComposition from "./BodyComposition.ts"

class BodyData {
    weight: number
    height: number
    bodyCircumferences: BodyCircumferences
    bodyComposition: BodyComposition
    skinFolds: Map<string, number> | null = null

    constructor(weight: number, height: number, bodyCircumferences: BodyCircumferences, bodyComposition: BodyComposition, skinFolds: Map<string, number> | null) {
        this.weight = weight
        this.height = height
        this.bodyCircumferences = bodyCircumferences
        this.bodyComposition = bodyComposition
        this.skinFolds = skinFolds
    }
}
export default BodyData
